sgrep/bin/main_sgrep.ml

```diff
@@ -540,10 +566,14 @@ let main () =
     (* main entry *)
     (* --------------------------------------------------------- *)
     | x::xs -> 
-        if !rules_file <> ""
-        then sgrep_with_rules !rules_file (x::xs)
-        else sgrep_with_one_pattern (x::xs)
-
+        (try (
+          if !rules_file <> ""
+          then sgrep_with_rules !rules_file (x::xs)
+          else sgrep_with_one_pattern (x::xs)
+        ) with exn -> 
+          pr (format_output_exception exn); (* todo, should be pr2 probably *)
```

Comment:
cc @aryx . What is the future of the sgrep CLI? Would it be reasonable to default to JSON output in all cases? Simplifies things for sgrep_lint but I don't have full visibility.

Reply:
I like to have a simple path to test sgrep core functionality. When there's a bug, I prefer to limit as much as possible layers, so I just want one pattern (no yaml file, no sgrep-lint wrapper) and a test file. Before it was bad because there was sgrep_lint.ml and sgrep_generic.ml but now it's mostly the same code behind -e and -rules_file (they both call the same sgrep_generic.ml) so I think it's fine.